libsemigroups_cppyy/__init__.py

```diff
@@ -18,12 +18,36 @@
 import os
 import pkgconfig
 from packaging import version
+import re
 
 
 def minimum_libsemigroups_version():
     return "1.0.7"
 
 
+def compare_version_numbers(supplied, required):
+    "Returns True if supplied >= required"
+
+    if isinstance(supplied, str) and isinstance(required, str):
+        return version.parse(supplied) >= version.parse(required)
+    else:
+        raise TypeError("expected a string, got a " + vers.__name__)
+
+
+def libsemigroups_version():
+    # the try-except is require pkgconfig v1.5.0 which is very recent, and
+    # hence not on conda at time of writing.
+    try:
+        v = pkgconfig.modversion("libsemigroups")
+    except AttributeError:
+        # this is just the guts of the modversion method in pkgconfig v1.5.1
+        v = pkgconfig.pkgconfig._query("libsemigroups", "--modversion")
+    if re.search("\d+.\d+.\d+-\d+-\w{7}", v):
```

Comment:
I think that with regular expressions, `.` matches any character apart from newline, whereas `\.` matches the literal dot character. This maybe isn't even worth changing, but I suppose if `v` by some error was something like `1a2a3-1-abcdefg`, then this would no longer be able to slip through.
```suggestion
    if re.search("\d+\.\d+\.\d+-\d+-\w{7}", v):
```